core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -1840,10 +1843,9 @@ public void doDeleteUser(String userName) throws UserStoreException {
             }
             dbConnection.commit();
         } catch (SQLException e) {
+            DatabaseUtil.rollBack(dbConnection);
             String msg = "Error occurred while deleting user : " + userName;
-            if (log.isDebugEnabled()) {
-                log.debug(msg, e);
-            }
+            log.error(msg, e);
```

Comment:
Can we add a comment here stating the reason for both logging and throwing the exception?

Reply:
Fixed https://github.com/wso2/carbon-kernel/pull/2810/commits/66bfa2016cccc221cb3ad7fbcd1a354bd6ab6cb2